glslc/test/environment.py

```diff
@@ -62,11 +63,16 @@ def write(self, parent):
 
 class File:
     """Specifies a file by name and content."""
-    def __init__(self, name, content):
+    def __init__(self, name, content, readable=True):
         self.name = name
         self.content = content
+        self.readable = readable
 
     def write(self, directory):
         """Writes content to directory/name."""
-        with open(os.path.join(directory, self.name), 'w') as f:
+        full_path = os.path.join(directory, self.name)
+        with open(full_path, 'w') as f:
             f.write(self.content)
+        if not self.readable:
+          access_bits = stat.S_IMODE(os.stat(full_path).st_mode)
+          os.chmod(full_path, access_bits & ~stat.S_IREAD)
```

Comment:
I think I should change it to S_IRUSR, gnu says:
"S_IREAD is an obsolete synonym provided for BSD compatibility."